Assets/Plugins/Source/Editor/Windows/LogLevelWindow.cs

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2024 PlayEveryWare
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+using UnityEngine;
+using UnityEditor;
+using System;
+using System.Collections.Generic;
+
+#if !EOS_DISABLE
+namespace PlayEveryWare.EpicOnlineServices.Editor.Windows
+{
+    public class LogLevelWindow : EOSEditorWindow
+    {
+        LogLevelConfig currentLogLevelConfig;
+
+        string[] categories = LogLevelUtility.LogCategoryStringArray;
+        string[] levels = LogLevelUtility.LogLevelStringArray;
+
+        int selectedCategoryIndex = 0;
+        bool showAllCategories = false;
+        string previousAllCategoryLogLevel;
+
+        [MenuItem("Tools/EOS Plugin/Log Level")]
+        public static void ShowWindow()
+        {
+            GetWindow<LogLevelWindow>("Log Level");
+        }
+        protected override void Setup()
+        {
+            currentLogLevelConfig = EpicOnlineServices.Config.Get<LogLevelConfig>();
+
+            // Config.Get<T> only guarantees T to be not null, but not for its members
+            if (currentLogLevelConfig.logCategoryLevelPairs == null) 
+            {
+                currentLogLevelConfig.logCategoryLevelPairs = new List<LogCategoryLevelPair>();
+            }
+
+            // Initialize the list if config categories does not match the SDK categories, which might happen when using a different SDK version
+            if (currentLogLevelConfig.logCategoryLevelPairs.Count != categories.Length)
```

Comment:
This will not catch circumstances where a category is renamed, or if between one SDK and another there is one log category removed and one log category added.

Reply:
That is true, but will be ultra rare, since it only happens with a change from the SDK.
The category names are only used when creating the json, and does not check the names when reading,
so rename won't be a problem.
removing and adding a category is even more rare, but in those case when it happens, the categories and levels are still still represented correctly in the UI.

The purpose of this is to make sure there are not any out of range exceptions

Reply:
That's a good point regarding the intent of the check. It does however reveal that the aforementioned circumstance can happen and is not accounted for by the current implementation. If this circumstance _did_ happen (and I completely agree it would be extraordinarily rare) I can see it causing a fair amount of confusion and subsequent mayhem.